contracts/TopShot.cdc

```diff
@@ -654,10 +665,95 @@ pub contract TopShot: NonFungibleToken {
+                    let name = self.getEditionName()
+                    let max = TopShot.getNumMomentsInEdition(setID: self.data.setID, playID: self.data.playID) ?? 0;
+                    let editionInfo = MetadataViews.Edition(name: name, number: UInt64(self.data.serialNumber), max: UInt64(max))
+                    let editionList: [MetadataViews.Edition] = [editionInfo]
+                    return MetadataViews.Editions(
+                        editionList
+                    )
+                case Type<MetadataViews.Serial>():
+                    return MetadataViews.Serial(
+                        UInt64(self.data.serialNumber)
+                    )
+                case Type<MetadataViews.Royalties>():
+                    return MetadataViews.Royalties(
+                        royalties: [
+                            // No royalty for Moments yet defined for 3rd party marketplaces
+                        ]
+                    )
+                case Type<MetadataViews.ExternalURL>():
+                    return MetadataViews.ExternalURL(self.getMomentURL())
+                case Type<MetadataViews.NFTCollectionData>():
+                    return MetadataViews.NFTCollectionData(
+                        storagePath: /storage/MomentCollection,
+                        publicPath: /public/MomentCollection,
+                        providerPath: /private/MomentCollection,
+                        publicCollection: Type<&TopShot.Collection{TopShot.MomentCollectionPublic}>(),
+                        publicLinkedType: Type<&TopShot.Collection{TopShot.MomentCollectionPublic,NonFungibleToken.Receiver,NonFungibleToken.CollectionPublic,MetadataViews.ResolverCollection}>(),
+                        providerLinkedType: Type<&TopShot.Collection{NonFungibleToken.Provider}>(),
+                        createEmptyCollectionFunction: (fun (): @NonFungibleToken.Collection {
+                            return <-TopShot.createEmptyCollection()
+                        })
+                    )
+                case Type<MetadataViews.NFTCollectionDisplay>():
+                    let media = MetadataViews.Media(
+                        file: MetadataViews.HTTPFile(
+                            url: "https://nbatopshot.com/static/img/top-shot-logo-horizontal-white.svg"
+                        ),
+                        mediaType: "image/svg+xml"
+                    )
+                    return MetadataViews.NFTCollectionDisplay(
+                        name: "NBA TopShot",
+                        description: "NBA Top Shot is your chance to own, sell, and trade official digital collectibles of the NBA and WNBA's greatest plays and players",
+                        externalURL: MetadataViews.ExternalURL("https://nbatopshot.com"),
+                        squareImage: media,
+                        bannerImage: media,
+                        socials: {}
+                    )
+                case Type<MetadataViews.Traits>():
```

Comment:
is there a particular reason why the rest of the traits from TopShotMomentMetadataView are not added here? If you want thirdparty tools to be able to read this data it is a lot easier for them if you add all of them here.

Reply:
Views that are specific to a producer solution like TopShotMomentMetadataView are really meant for internal consuption. Since using them for external parties will lead to tight couplings and hardcoding rather then exploring and adapting.

Reply:
The only concern I would have with adding traits for all current fields of TopShotMomentMetadataView would be some of them are optional/will soon be optional (In the near future we may have plays that have no specific player in the metadata, instead the play would be attributed t a whole team). Would there be any concerns with having traits present for some moments and not others?

Reply:
Traits don't have to be uniform! Think of whether a rookie moment should be a trait or not as an example. Putting a trait for the absence of something on that many moments feels like a waste. All that traits are telling you is what this NFT has, not what the complete of traits in an entire collection can be

Reply:
Definitely going to follow up on this soon! Just merging this PR for now to a dev/feature branch to enable others to more easily contribute.